File: day1/day1_solution_part2.go
package main

import (
	"encoding/csv"
	"fmt"
	"os"
	"strconv"
)

var sampleInput = []int{1, -2, 3, 1,}

func retrieveFrequencies(filepath string) [][]string {
	var (
		file, _   = os.Open(filepath)
		csvreader = csv.NewReader(file)
	)

	contents, _ := csvreader.ReadAll()

	fmt.Println(contents)
	return contents
}

//func compareFrequencies(frequencies []int) {
func compareFrequencies(frequencies[][]string) {
	var (
		dup       bool
		lastValue int

		sum =  make(map[int]string)
	)

	for dup != true {
		for _, f := range frequencies {

			fc, _ := strconv.Atoi(f[0])
			lastValue += fc

			if  _, ok := sum[lastValue]; ok == false {
				sum[lastValue] = ""
				continue
			}

			fmt.Println(lastValue)
			dup = true
			break
		}
	}
}

func main() {
	filepath := os.Getenv("AOC_INPUT_FILE")
	compareFrequencies(retrieveFrequencies(filepath))

	//compareFrequencies(sampleInput)
}

File: day1/day1_solution_part2.py
import csv
import os 
import subprocess

sample_input = [1, -2, 3, 1]

#Opens and retrieves input
def retrieve_frequencies(filepath):
    with open(filepath, 'r') as frequencies:
        reader = csv.reader(frequencies)
        file_contents = list(reader)

    return file_contents

def compare_frequencies():
    freq = retrieve_frequencies(filepath)
    dups = False
    sums = {}
    vals = [0]

    while dups == False:
        for f in freq:
            #ans = vals[-1] + f
            ans = vals[-1] + int(f[0])
            vals.append(ans) 
            
            if ans in sums:
                print(ans)
                dups = True
                break
            else:
                sums[ans] = ""

if __name__ == "__main__":
    filepath = os.getenv("AOC_INPUT_FILE")
    compare_frequencies()
    
